Common/Util/CurrencyPairUtil.cs

```diff
@@ -0,0 +1,182 @@
+            /// The two currency pairs are the inverse of each other
+            /// </summary>
+            InverseMatch
+        }
+
+        /// <summary>
+        /// Returns how two currency pairs are related to each other
+        /// </summary>
+        /// <param name="pairA">The first pair</param>
+        /// <param name="pairB">The second pair</param>
+        /// <returns>The <see cref="Match"/> member that represents the relation between the two pairs</returns>
+        public static Match ComparePair(this Symbol pairA, Symbol pairB)
+        {
+            if (pairA == pairB)
+            {
+                return Match.ExactMatch;
+            }
+
+            string baseCurrencyA;
+            string quoteCurrencyA;
+
+            DecomposeCurrencyPair(pairA, out baseCurrencyA, out quoteCurrencyA);
+
+            string baseCurrencyB;
+            string quoteCurrencyB;
+
+            DecomposeCurrencyPair(pairB, out baseCurrencyB, out quoteCurrencyB);
+
+            if (baseCurrencyA == quoteCurrencyB && baseCurrencyB == quoteCurrencyA)
+            {
+                return Match.InverseMatch;
+            }
+
+            return Match.NoMatch;
+        }
+
+        /// <summary>
+        /// Returns how two currency pairs are related to each other
+        /// </summary>
+        /// <param name="pairA">The first pair</param>
+        /// <param name="baseCurrencyB">The base currency of the second pair</param>
+        /// <param name="quoteCurrencyB">The quote currency of the second pair</param>
+        /// <returns>The <see cref="Match"/> member that represents the relation between the two pairs</returns>
+        public static Match ComparePair(this Symbol pairA, string baseCurrencyB, string quoteCurrencyB)
+        {
+            if (pairA.Value == baseCurrencyB + quoteCurrencyB)
+            {
+                return Match.ExactMatch;
+            }
+
+            string baseCurrencyA;
+            string quoteCurrencyA;
+
+            DecomposeCurrencyPair(pairA, out baseCurrencyA, out quoteCurrencyA);
+
+            if (baseCurrencyA == quoteCurrencyB && baseCurrencyB == quoteCurrencyA)
+            {
+                return Match.InverseMatch;
+            }
+
+            return Match.NoMatch;
+        }
+
+        /// <summary>
+        /// Returns whether a currency pair contains a certain currency as base or as quote
+        /// </summary>
+        /// <param name="pair">The currency pair to check the sides of</param>
+        /// <param name="currency">The currency to look for</param>
+        /// <returns>True if currency is the base or quote currency of pair, false if not</returns>
+        public static bool PairContainsCurrency(this Symbol pair, string currency)
+        {
+            string baseCurrency;
+            string quoteCurrency;
+
+            DecomposeCurrencyPair(pair, out baseCurrency, out quoteCurrency);
+
+            return baseCurrency == currency || quoteCurrency == currency;
```

Comment:
Replaced with `baseCurrency.Equals(currency, StringComparison.InvariantCultureIgnoreCase)` because I think it's prettier than `string.Compare(...) == 0`, and added some tests for it